### PR TITLE
#401: add support for `validation_packages.yml`

### LINKED ISSUE #401: [Feature Request] Support validation_packages.yml
This week will see a rollout of the new [Continuous Quality Control pipelines](https://github.com/nfdi4plants/ARC-specification/blob/main/ARC%20specification.md#mechanisms-for-arc-quality-control) on the DataHUB. This means that all ARCs using the old validation pipelines will need an update that moves them to using the new `validation_packages.yml` config file to opt-in for user-selected validation packages.

While [POCs of this exist across the ARCtrl stack]() and a manual migration guide will be uploaded, the ARCitect as the central tool must support this feature to some extent on launch. This will be on 2024-07-10, so time is of essence.

Therefore, ARCtrl needs:
- a Data model for validation packages
- Reader/Writer contracts for the file
- a Top-level API

Implementation details:

-  use the YAMLicious package and transpilable encoder/decoder logic for the yml file
-  a new `ARCtrl.ValidationPackages` project that contains a type model and low-level APIs

### PR DESCRIPTION
**Pull request by kMutagene**
This PR
- [x] Adds a type model for validation packages and the `validation_packages.yml` file [according to spec]()
- [x] Adds YAMLicious-based Encoder/Decoder logic
- [x] Adds necessary contracts
~~- [ ] Adds top-level convenience APIs~~
~~- [ ] tests~~
  ~~- [ ] type model~~
  ~~- [ ] read/write yaml~~
  ~~- [ ] contracts~~
  ~~- [ ] API~~

### CODE REVIEW

File: src/ValidationPackages/ValidationPackage.fs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        with get() = _name
+        and set(name) = _name <- name
+
+    member this.Version
+        with get() = _version
+        and set(version) = _version <- version
+
+    static member make name version = ValidationPackage(name=name, ?version=version)
+
+    static member create(name,?version) : ValidationPackage =
```
Comment (Freymaurer) [inline src/ValidationPackages/ValidationPackage.fs:22]: You can remove this, it is duplicating primary constrcutor logic

File: src/ValidationPackages/ValidationPackagesConfig.fs [hunk 2]
```diff
@@ -0,0 +1,42 @@
+﻿namespace ARCtrl.ValidationPackages
+
+open ARCtrl.Helper
+open Fable.Core
+
+[<AttachMembers>]
+type ValidationPackagesConfig(validation_packages, ?arc_specification) =
+    
+    let mutable _validation_packages : ValidationPackage array = validation_packages
```
Comment (Freymaurer) [inline src/ValidationPackages/ValidationPackagesConfig.fs:9]: use resizearray (translated to native js/py arrays

File: src/ValidationPackages/ValidationPackagesConfig.fs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        with get() = _validation_packages
+        and set(validation_packages) = _validation_packages <- validation_packages
+
+    member this.ARCSpecification
+        with get() = _arc_specification
+        and set(arc_specification) = _arc_specification <- arc_specification
+
+    static member make validation_packages arc_specification = ValidationPackagesConfig(validation_packages=validation_packages, ?arc_specification=arc_specification)
+
+    static member create(validation_packages,?arc_specification) : ValidationPackagesConfig =
```
Comment (Freymaurer) [inline src/ValidationPackages/ValidationPackagesConfig.fs:22]: remove, same as above

File: src/ValidationPackages/ValidationPackagesConfig.fs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        ValidationPackagesConfig.make validation_packages arc_specification
+    
+    static member toString (vp : ValidationPackagesConfig) =
```
Comment (Freymaurer) [inline src/ValidationPackages/ValidationPackagesConfig.fs:25]: This does not return string?

File: src/Yaml/ARCtrl.Yaml.fsproj
Comment (Freymaurer) [inline src/Yaml/ARCtrl.Yaml.fsproj:1]: missing fable files

**Review by Freymaurer — CHANGES_REQUESTED** [review]

File: src/ValidationPackages/ValidationPackagesConfig.fs (hunk 4 above)
Comment (kMutagene) [reply src/ValidationPackages/ValidationPackagesConfig.fs:25]: I was mirroring the API of Comment here: 
https://github.com/nfdi4plants/ARCtrl/blob/403081ee9648a87ffd0198baaebef025df51b006/src/Core/Comment.fs#L27-L28
But i do not need this function at all so i'll just remove it until it is needed

**Comment by kMutagene** [comment]
tests and top-level API postponed to another PR to avoid merge conflicts with another contribution @Freymaurer

File: src/Yaml/ARCtrl.Yaml.fsproj
Comment (Freymaurer) [reply src/Yaml/ARCtrl.Yaml.fsproj:1]: still missing

**Review by Freymaurer — CHANGES_REQUESTED** [review]

File: src/Yaml/ARCtrl.Yaml.fsproj
Comment (kMutagene) [reply src/Yaml/ARCtrl.Yaml.fsproj:1]: https://youtu.be/WcWM_1hBu_c?t=55

Comment (Freymaurer) [reply src/Yaml/ARCtrl.Yaml.fsproj:1]: > https://youtu.be/WcWM_1hBu_c?t=55

;)

**Review by Freymaurer — APPROVED** [review]
Nice!

**Pull request merged**